ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+
+### Introduction
+
+Different focus:
+
+  * Akka as a toolkit for building distributed systems, offering the full power but also exposing the inherent essential complexity.
+  
+  * Orleans restricts applicability in order to allow seamless use without understanding distributed computing.
+  
+#### Lifecycle
+
+  * Orleans Grains do not have a lifecycle, can neither be started nor stopped. Consequently they also cannot fail and be restarted, therefore Orleans does not offer tools for software failure handling—the failure handling aspects concentrate on recovering from hardware crashes.
+  
+  * Akka Actors implement the full model including defined lifecycle beginning and end, these are explicit operations. Persistent Actors support extending the lifecycle of a logical unit of computation beyond the lifecycle of the running process instance. Restarting an Actor provides powerful means for automatic service recovery.
+  
+#### Automatic Creation
+
+  * Orleans Grains are created automatically whenever needed, which implies that initialization activities cannot have persistent effects—otherwise these would be uncontrollable. OTOH this frees the user from having to consider the need for Grain creation. It also means that asking for a Grain you don’t know which implementation you will get.
+  
+  * Akka actors are explicitly created by their parent, implementing mandatory parental supervision. This allows precise control over when initialization actions are performed and which exact actor class is being created.
+  
+#### Virtual Actor Space
+
+  * In Orleans each type of Grain corresponds to a practically infinite space of Grain instances that conceptually all exist from the beginning of the universe to its end. The relation to the physical Actors that implement the Grains is explained similar to virtual and physical memory, but this comparison is misleading since the virtual address space of a process is explicitly populated with the desired contents instead of containing the whole system’s information by default.
+  
+  * Akka’s explicit lifecycle management requires that all currently running Actors must have been explicitly started in the past and will have to be explicitly stopped in the future. Higher-level abstractions like ClusterSharding provide the ability to opt into an Orleans-like model where instances can be addressed by their logical name instead of a concrete ActorRef, but the API for accessing them is provided by other Actors that perform the necessary lookups and initialization on behalf of the user.
+
+#### Automatic Actor Placement and Load Balancing
+
+  * Orleans Grains are deployed on silos that can span multiple cluster nodes and the user has no influence on their precise placement or load-dependent movement. This means that making use of elastically provisioned resources is fully automatic and built into the model.
```

Comment:
This is correct, although we do gradually add more complex policies to impact how load balancing is done. For example, we now have a policy to impact grain placement to balance the number of actors per server (https://github.com/dotnet/orleans/blob/master/src/Orleans/Placement/ActivationCountBasedPlacement.cs).
The idea is that the developer will provide the desired policy (with default policy being a good and reasonable one that handles the majority of cases), but will not need to implement the policy itself. The difficulty in implementing the policy is actually quiet significant: distributed load balancing algorithm has to be robust to fluctuations, spikes, thrashing....most of the programmers don't want to deal with those details. 
We do add more and more polices, to provide more of that "customability and extensibility" I mentioned before. Again, "guide the developers down a path of best practices principle".